hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/simple/HoodieGlobalSimpleIndex.java

```diff
@@ -72,85 +68,37 @@ public <R> HoodieData<HoodieRecord<R>> tagLocation(
   protected <R> HoodieData<HoodieRecord<R>> tagLocationInternal(
       HoodieData<HoodieRecord<R>> inputRecords, HoodieEngineContext context,
       HoodieTable hoodieTable) {
-
-    HoodiePairData<String, HoodieRecord<R>> keyedInputRecords =
-        inputRecords.mapToPair(entry -> new ImmutablePair<>(entry.getRecordKey(), entry));
-    HoodiePairData<HoodieKey, HoodieRecordLocation> allRecordLocationsInTable =
-        fetchAllRecordLocations(context, hoodieTable, config.getGlobalSimpleIndexParallelism());
-    return getTaggedRecords(keyedInputRecords, allRecordLocationsInTable, hoodieTable);
+    List<Pair<String, HoodieBaseFile>> latestBaseFiles = getAllBaseFilesInTable(context, hoodieTable);
```

Comment:
we could move L 71 within fetchRecordGlobalLocations() right?

Reply:
+1